pkg/deploy/factory.go

```diff
@@ -92,3 +105,33 @@ func (f Factory) globalKappDeployRawOpts() []string {
 	}
 	return nil
 }
+
+// GetClusterVersion returns the kubernetes API version for the cluster which has been supplied to kapp-controller via a kubeconfig
+func (f Factory) GetClusterVersion(saName string, specCluster *v1alpha1.AppCluster, objMeta *metav1.ObjectMeta, log logr.Logger) (*version.Info, error) {
+	// this logic is duplicated here and in ProcessOpts: if the serviceAccount name is present, we will be deploying to this cluster
+	if len(saName) > 0 {
```

Comment:
we should use a switch statement as above to catch the case when it's neither saName or kubeconfig